### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by yuvipanda**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+   
+   ```bash
+   python3 -m venv .
+   source bin/activate
+   ```
+   
+2. Do a dev install of the `tljh_voila_gallery` package.
+
+   ```bash
+   pip install -e tljh_voila_gallery
```
Comment (jtpio) [inline CONTRIBUTING.md]: Shouldn't this be the path to the package?

```suggestion
   pip install -e tljh-voila-gallery
```

File: tljh-voila-gallery/tljh_voila_gallery/__init__.py [hunk 2]
```diff
@@ -5,6 +5,11 @@
 from pkg_resources import resource_stream, resource_filename
 from ruamel.yaml import YAML
 from tljh.hooks import hookimpl
```
Comment (jtpio) [inline tljh-voila-gallery/tljh_voila_gallery/__init__.py:7]: This still requires TLJH to be installed locally?

**Comment by yuvipanda** [comment]
Good catch, @jtpio. Addressed both your comments now.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+   ```bash
+   python3 -m tljh_voila_gallery.build_images
+   ```
+
+   This requires docker be available and usable from your current
+   user without root. If you need to use `sudo` before your `docker`
+   calls, you should use `sudo $(which python3)` rather than just
+   `python3`
+
+4. Start JupyterHub with the testing config
```
Comment (jtpio) [inline CONTRIBUTING.md]: We also need `configurable-http-proxy` before starting JupyterHub. And node as well unless it's available via other channels?

Comment (yuvipanda) [reply CONTRIBUTING.md]: Ah, true! I already had them installed I think so didn't explicitly add them. I'll add pointers to both...

Comment (yuvipanda) [reply CONTRIBUTING.md]: Added!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# Contributing
+
+## Setting up a dev environment
+
+Though voila-gallery is deployed as a [TLJH](https://tljh.jupyter.org)
+plugin, you can develop it locally without needing TLJH. You just need
```
Comment (jtpio) [inline CONTRIBUTING.md]: Then TLJH is still required? (at least as a dev dependency)

Comment (yuvipanda) [reply CONTRIBUTING.md]: Yep it is, mostly because we're importing it unconditionally in a bunch of places.

Comment (yuvipanda) [reply CONTRIBUTING.md]: It won't do anything to your system though... And we *could* remove it if needed. I'm ok with leaving it in though.

Comment (jtpio) [reply CONTRIBUTING.md]: ok we can keep it. Then we can remove this part: "you can develop it locally without needing TLJH".

Comment (yuvipanda) [reply CONTRIBUTING.md]: I clarified it to mean you don't have to *install* TLJH, which takes forever and shouldn't be done on your own system. Is that clearer?

Comment (jtpio) [reply CONTRIBUTING.md]: yes!

Comment (yuvipanda) [reply CONTRIBUTING.md]: Awesome :) <3

**Comment by jtpio** [comment]
Went through the whole setup and started jupyterhub locally.
However `http://localhost:8000/services/gallery` is giving back 404.

Haven't investigated why yet but maybe I'm missing something?

Logs say the service is running:

```
[D 2019-06-17 10:47:55.083 JupyterHub proxy:765] Proxy: Fetching POST http://127.0.0.1:8001/api/routes/
[I 2019-06-17 10:47:55.084 JupyterHub proxy:242] Adding service gallery to proxy /services/gallery/ => http://127.0.0.1:9888
[D 2019-06-17 10:47:55.085 JupyterHub proxy:765] Proxy: Fetching POST http://127.0.0.1:8001/api/routes/services/gallery
10:47:55.088 [ConfigProxy] info: Adding route / -> http://x1:8081
10:47:55.088 [ConfigProxy] info: Route added / -> http://x1:8081
10:47:55.089 [ConfigProxy] info: 201 POST /api/routes/
10:47:55.089 [ConfigProxy] info: Adding route /services/gallery -> http://127.0.0.1:9888
10:47:55.089 [ConfigProxy] info: Route added /services/gallery -> http://127.0.0.1:9888
10:47:55.090 [ConfigProxy] info: 201 POST /api/routes/services/gallery
[I 2019-06-17 10:47:55.090 JupyterHub app:2422] JupyterHub is now running at http://:8000
WARNING:tornado.access:404 GET /services/gallery (127.0.0.1) 0.55ms
[D 2019-06-17 10:48:55.094 JupyterHub app:1812] Managed service gallery running at http://127.0.0.1:9888
[D 2019-06-17 10:49:55.094 JupyterHub app:1812] Managed service gallery running at http://127.0.0.1:9888

```

File: CONTRIBUTING.md (hunk 3 above)
Comment (jtpio) [reply CONTRIBUTING.md]: Thanks!

**Comment by yuvipanda** [comment]
@jtpio try http://localhost:8000/services/gallery/ (trailing slash). That's fixed in #26

**Comment by jtpio** [comment]
That did it, thanks.

All good on my side, let's merge!

**Comment by yuvipanda** [comment]
Thanks for the review + merge, @jtpio!

**Pull request merged**